src/protocols/ethernet-10G-KR.stanza

```diff
@@ -0,0 +1,146 @@
+
+## 10GBASE-KR Blocking Capacitors
+
+The 10GBASE-KR specification calls for AC coupling for the data lanes. This is typically
+achieved using a blocking capacitor. When connecting two active components, this typically
+means blocking caps from `Tx -> Rx` on both sides of the link. When connecting an
+active component to a passive component, this typically means adding the blocking caps only on
+the `Tx -> Rx` side of the link.
+
+The functions in this module allow you to pass a blocking capacitor as an `Instantiable`.
+This component will get instantiated for each leg of the diff-pair. These functions handle
+the topology configuration, but the user needs to set a `pin-model` statement on the capacitor component.
+
+
+<DOC>
+#use-added-syntax(jitx)
+defpackage jsl/protocols/ethernet-10G-KR:
+  import core
+  import jitx
+  import jitx/commands
+
+  import jsl/bundles
+  import jsl/errors
+  import jsl/si/helpers
+  import jsl/si/couplers
+  import jsl/si/pairs
+  import jsl/pin-assignment
+  import jsl/si/constraints
+
+doc: \<DOC>
+@brief 10GBASE-KR Bundle
+The Data Bundle consists of a number (1 or more) of RX and TX pairs (1 RX/TX pair set = 1 lane)
+@param num-lanes The number of lanes to be configured for this bundle.
+<DOC>
+public pcb-bundle ethernet-10G-KR (num-lanes:Int) :
+  name = "10GBASE-KR"
+  description = "10GBASE-KR Ethernet Communications Link"
+  port lane : lane-pair[num-lanes]
+
+doc: \<DOC>
+@brief Construct the 10GBASE-KR Topology and Constraints
+
+This function constructs the 10GBASE-KR topology and applies constraints
+to the channel for intra-pair skew and channel loss.
+
+@param cons The Constraint structure to be used for the routing.
+@param cap-src-dst Optional DC blocking capacitor to be inserted between the src.TX and dst.RX signal ends. Note that
+there are 2 blocking caps inserted, one on each of the pair of P,N signals. The capacitor needs to have pin models applied to its pins
+in order for the overall skew and loss values to be respected.
+@param cap-dst-src Optional DC blocking capacitor to be inserted between the dst.TX and src.RX signal ends. This capacitor
+covers the reverse direction connection between dst.TX and src.RX. If the second component was a connector, then typically
+no capacitor is inserted between dst.TX and src.RX (cap-dst-src = false).
+@param src This is expected to be a port of `Bundle` type `ethernet-10G-KR`
+@param dst This is expected to be a port of `Bundle` type `ethernet-10G-KR`
+@param null-modem If set to true, connects src.TX through the optional cap-src-dst to dst.RX and dst.TX through the optional cap-dst-src to src.RX. 
+If set to false, connects src.TX through the optional cap-src-dst to dst.TX and dst.RX through the optional cap-dst-src to src.RX
+<DOC>
+public defn connect-10G-KR (cons, cap-src-dst:Instantiable|False, cap-dst-src:Instantiable|False, src:JITXObject, dst:JITXObject, null-modem:True|False) :
```

Comment:
`cons` is lacking a type. It should be `SI-Constraint`

You might want to consider using a keyword arg for `null-modem` as a random true/false at the end of the function is going to be hard to read. 

---

I don't agree with this approach to defining constraints on topologies.

1.  Repetition - notice how every block of the match has very similar statements
2.  Has many arguments - for me, this is a leading indicator that the structure is wrong.
3. Not Extensible - it covers cases that may never happen, but also is not flexible enough to give the user options. This makes this function very brittle. 

I know we want to make it seem like magic, but this isn't magic to me. This is akin to a reference design. It is one instance of making this work - it is not an expressive system from which I can make any 10G-KR design. 

I also know we need to get this out. I think we are still in the alpha stage and so in my opinion we can make API changes as we want. This function also doesn't stop me or the end user from using what I think is the "right" way to design this - so I think we should move forward in service of haste.